apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/[contractAddress]/modules/components/Claimable.tsx

```diff
@@ -0,0 +1,656 @@
+  pricePerUnit: bigint;
+  currency: string;
+  maxMintPerWallet: bigint;
+  startTimestamp: number;
+  endTimestamp: number;
+  auxData: string;
+};
+
+function ClaimableModule(props: ModuleInstanceProps) {
+  const { contract, ownerAccount } = props;
+  const account = useActiveAccount();
+
+  const primarySaleRecipientQuery = useReadContract(
+    ClaimableERC721.getSaleConfig,
+    {
+      contract: contract,
+    },
+  );
+  const claimConditionQuery = useReadContract(
+    ClaimableERC721.getClaimCondition,
+    {
+      contract: contract,
+    },
+  );
+
+  const isErc721 = props.contractInfo.name === "ClaimableERC721";
+  const isSequentialTokenIdInstalled = !!props.allModuleContractInfo.find(
+    (module) => module.name.includes("SequentialTokenId"),
+  );
+
+  const mint = useCallback(
+    async (values: MintFormValues) => {
+      if (!account) {
+        throw new Error("Wallet not connected");
+      }
+
+      let mintTx: PreparedTransaction;
+      if (isErc721) {
+        mintTx = ClaimableERC721.mint({
+          contract,
+          to: values.recipient,
+          quantity: values.quantity,
+        });
+      } else if (values.tokenId) {
+        mintTx = ClaimableERC1155.mint({
+          contract,
+          to: values.recipient,
+          quantity: values.quantity,
+          tokenId: BigInt(values.tokenId),
+        });
+      } else {
+        throw new Error("Invalid tokenId");
+      }
+
+      await sendAndConfirmTransaction({
+        account: account,
+        transaction: mintTx,
+      });
+    },
+    [contract, account, isErc721],
+  );
+
+  const update = useCallback(
+    async (values: ConfigFormValues) => {
+      if (!ownerAccount) {
+        throw new Error("Not an owner account");
+      }
+
+      console.log("values: ", values);
+
+      let setClaimConditionTx: PreparedTransaction;
+      if (isErc721) {
+        setClaimConditionTx = ClaimableERC721.setClaimCondition({
+          ...values,
+          contract,
+          allowList:
+            values.allowList && values.allowList.length > 0
+              ? values.allowList.map(({ address }) => address)
+              : undefined,
+        });
+      } else if (values.tokenId) {
+        setClaimConditionTx = ClaimableERC1155.setClaimCondition({
+          ...values,
+          contract,
+          allowList:
+            values.allowList && values.allowList.length > 0
+              ? values.allowList.map(({ address }) => address)
+              : undefined,
+          tokenId: BigInt(values.tokenId),
+        });
+      } else {
+        throw new Error("Invalid tokenId");
+      }
+
+      await sendAndConfirmTransaction({
+        account: ownerAccount,
+        transaction: setClaimConditionTx,
+      });
+
+      const setSaleConfigTx = ClaimableERC721.setSaleConfig({
+        contract: contract,
+        primarySaleRecipient: values.primarySaleRecipient,
+      });
+
+      await sendAndConfirmTransaction({
```

Comment:
primary sales recipient and claim conditions should be two separate mutations and should two separate sections in UI